File: mybatis-plus-sample-performance-analysis/src/main/java/com/baomidou/samples/performance/config/MybatisPlusConfig.java
package com.baomidou.samples.performance.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import com.baomidou.mybatisplus.extension.plugins.PerformanceInterceptor;

/**
 * MybatisPlus配置文件
 * @author nieqiurong 2018/8/11 21:10.
 */
@Configuration
public class MybatisPlusConfig {


    @Bean
    public PerformanceInterceptor performanceInterceptor(){
        //启用性能分析插件
        return new PerformanceInterceptor();
    }


}

File: mybatis-plus-sample-kotlin/src/main/kotlin/com/baomidou/mybatisplus/samples/kotlin/config/MybatisPlusConfig.kt
package com.baomidou.mybatisplus.samples.kotlin.config

import com.baomidou.mybatisplus.extension.plugins.inner.PaginationInnerInterceptor
import org.mybatis.spring.annotation.MapperScan
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

/**
 * @author nieqiurong
 */
@Configuration
@MapperScan("com.baomidou.mybatisplus.samples.kotlin.mapper")
open class MybatisPlusConfig {

    @Bean
    open fun paginationInnerInterceptor(): PaginationInnerInterceptor {
        return PaginationInnerInterceptor();
    }

}